tests/PHPStan/Analyser/nsrt/bug-11311.php

```diff
@@ -65,3 +65,17 @@ function bug11331c(string $url):void {
 		assertType('array{string, string|null, string|null, string, string}', $matches);
 	}
 }
+
+class UnmatchedAsNullWithTopLevelAlternation {
+	function doFoo(string $s): void {
+		if (preg_match('/Price: (?:(£)|(€))\d+/', $s, $matches, PREG_UNMATCHED_AS_NULL)) {
+			assertType('array{string, string|null, string|null}', $matches); // could be array{0: string, 1: null, 2: string}|array{0: string, 1: string, 2: null}
```

Comment:
Yeah that would be nicer for sure ;) But strictly speaking at least this is correct.